version/distro/distro.go

```diff
@@ -147,3 +152,25 @@ func DSMVersion() int {
 		return v
 	})
 }
+
+func isUDMPro() bool {
+	if exists, err := fileContainsString("/etc/board.info", "UDMPRO"); err == nil && exists {
+		return true
+	}
+	if exists, err := fileContainsString("/etc/board.info", "Dream Machine PRO"); err == nil && exists {
+		return true
+	}
+	if exists, err := fileContainsString("/sys/firmware/devicetree/base/soc/board-cfg/id", "udm pro"); err == nil && exists {
+		return true
+	}
+	return false
+}
+
+func fileContainsString(filePath, searchString string) (bool, error) {
```

Comment:
optional: In case the file is large, using `lineread.File` to scan a line at a time _might_ be slightly nicer, but that doesn't solve the problem if the file is large but all one line, so I don't feel strongly about this. An `io.LimitedReader` could help with that, but then this starts to look complicated and I see `os.ReadFile` in use all over the place elsewhere.

I'm not particularly happy with any of the easily available options, so go with what you think is reasonable; leaving this how it is currently written is fine.

Reply:
I'm planning on leaving this because the files listed are known to be small in size. Although if someone changes this in the future, they may not realize how inefficient this is. Should I leave a comment alluding that if large files are ever used to refactor to a more efficient method of reading/parsing the files?